### PR TITLE
[RLlib; documentation] "RLlib in 60sec" overhaul.

### PR DESCRIPTION
**Pull request by sven1977**
"RLlib in 60sec" overhaul.
- Start moving rllib docs pages into "rllib" sub-folder (analogous to tune).
- Re-do `doc/source/rllib-index.rst` page (moved to `doc/source/rllib/index.rst`)
- Start moving code in the docs to their own "rllib/examples/documentation/" folder and CI-test all code snippets in the docs (this is work in progress). This PR tests the new example code in the "RLlib in 60sec" section.

TODO: In a follow up PR, we'll add sigils to the "Feature Overview" section, such that each feature of RLlib (e.g. multi-agent) has its own sigil. These will then also be usable in the examples scripts- and other sections of the new documentation.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -318,8 +318,9 @@ Papers
    :maxdepth: -1
    :caption: Ray RLlib
 
-   rllib.rst
+   rllib/index.rst
```
Comment (sven1977) [inline doc/source/index.rst:321]: Will start moving all rllib docs files into the `rllib` sub-folder to the rst's will no longer be in the top-leve directory. Analogous to how tune docs do this.

File: doc/source/rllib/index.rst [hunk 2]
```diff
@@ -0,0 +1,195 @@
+.. admonition:: We're hiring!
```
Comment (sven1977) [inline doc/source/rllib/index.rst:1]: This is the new RLlib main index page.

File: rllib/agents/trainer.py [hunk 3]
```diff
@@ -749,8 +749,9 @@ def env_creator_from_classpath(env_context):
         # Evaluation setup.
         self.evaluation_workers = None
         self.evaluation_metrics = {}
-        # Do automatic evaluation from time to time.
-        if self.config.get("evaluation_interval"):
+        # User would like to setup a separate evaluation worker set.
```
Comment (sven1977) [inline rllib/agents/trainer.py:752]: Small change in the code to make the new example (RLlib in 60s) work with fewer config-overrides.

File: rllib/examples/documentation/rllib_in_60s.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# __rllib-in-60s-begin__
```
Comment (sven1977) [inline rllib/examples/documentation/rllib_in_60s.py:1]: Script linked to from the documentation (new "RLlib in 60s" page)

File: doc/source/index.rst (hunk 1 above)
Comment (worldveil) [reply doc/source/index.rst:321]: good idea!

**Review by worldveil — APPROVED** [review]
LGTM!

**Pull request merged**